pkg/querier/queryrange/instrumentation.go

```diff
@@ -20,7 +20,7 @@ func InstrumentMiddleware(name string) Middleware {
 	return MiddlewareFunc(func(next Handler) Handler {
 		return HandlerFunc(func(ctx context.Context, req Request) (Response, error) {
 			var resp Response
-			err := instrument.TimeRequestHistogram(ctx, name, queryRangeDuration, func(ctx context.Context) error {
+			err := instrument.CollectedRequest(ctx, name, instrument.NewHistogramCollector(queryRangeDuration), instrument.ErrorCode, func(ctx context.Context) error {
```

Comment:
Could you explain me this change, please?

Reply:
`TimeRequestHistogram` is commented as deprecated.

Reply:
For even more context all I did was grab the implementation of what `TimeRequestHistogram` was doing and put it here.